canonical-web-and-design/maas-ui
# fix(ui): fix filtering nodes by tag

## Done

- Update the node filters to handle tag ids.

## QA

### MAAS deployment

To run this branch you will need access to one of the following MAAS deployments:

- [Karura](/HACKING.md#karura)
- [Bolla](/HACKING.md#bolla)
- [A development MAAS](/HACKING.md#development-deployment)
- [A local snap MAAS](/HACKING.md#snap-deployment) (this will not usually have machines)

### Running the branch

You can run this branch by:

- Serving with [dotrun](/HACKING.md#maas-ui-development-setup)
- [Building in a development MAAS](/HACKING.md#running-maas-ui-from-a-development-maas)

### QA steps

- Visit the machine list and filter dropdown and click on a filter.
- The machine list should update to show the correct machines.
- Repeat the same in the device list.
- Go to a controller config tab and click on a tag.
- You should get taken to the controller list which should be filtered by the tag.

## Fixes

Fixes: #3514.
